parquet-column/src/main/java/org/apache/parquet/column/MinMax.java

```diff
@@ -18,25 +18,17 @@
  */
 package org.apache.parquet.column;
 
-import java.util.Iterator;
-import java.util.Set;
-
-import org.apache.parquet.io.api.Binary;
 import org.apache.parquet.schema.PrimitiveComparator;
 
 /**
- * This class calculates the max and min values of a Set.
+ * This class calculates the max and min values of an iterable collection.
  */
 public final class MinMax<T> {
-  private PrimitiveComparator comparator;
-  private Iterator<T> iterator;
   private T min = null;
   private T max = null;
 
-  public MinMax(PrimitiveComparator comparator, Iterator<T> iterator) {
-    this.comparator = comparator;
-    this.iterator = iterator;
-    getMinAndMax();
+  public MinMax(PrimitiveComparator comparator, Iterable<T> iterable) {
```

Comment:
I think,  you should get warnings if you use a generic class without the generic type. `PrimitiveComparator<T>` or even `Comparator<T>` should work fine.

Reply:
Fixed. Thanks!